### PR TITLE
Allow outputting a newline at the end of files

### LINKED ISSUE #199: Add an option to leave a newline at the end of files
The POSIX standard for text files is that each line is "A sequence of zero or more non-\<newline\> characters plus a terminating \<newline\> character". Files output by svgcleaner currently don't contain a newline character (`\n`) at the end of the file, which breaks displaying the file with `cat` or similar in terminals, and shows warnings elsewhere (including GitHub).

For more information, check out this StackOverflow question: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

My proposal is to add a setting to append newlines to the end of each file. This adds exactly one byte to every file, so it makes sense to be disabled by default, due to the purpose of svgcleaner being to absolutely minify the SVG, yet allowing this would be great to avoid warnings and potential issues. This would probably go under the "Output" section of the `--help` page.

### PR DESCRIPTION
**Pull request by aaronfranke**
Fixes #199

This is my first time ever coding in Rust, so I apologize if the code is inefficient, but it does work.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -174,6 +174,14 @@ fn main() {
         }
     }
 
+    // Optionally add a newline to the end of the file.
+    // This is placed after the check for if the file is smaller. It's OK if the
+    // file grows when adding a newline, since the user explicitly wanted that.
+    let newline_flag = args.value_of(KEYS[Key::OutputNewline]).unwrap();
```
Comment (RazrFalcon) [inline src/main.rs]: I think we can use `cleaning_opt.output_newline` here. No need to parse it again.

File: data/help.txt [hunk 2]
```diff
@@ -80,6 +80,7 @@ Paths:
 
 Output:
     --trim-colors <FLAG>                    Use #RGB notation [default: yes]
+    --output-newline <FLAG>                 Output a newline at the end of the file [default: no]
```
Comment (RazrFalcon) [inline data/help.txt]: I think we should prefer `--append-newline`.

File: docs/images/after/output-newline.svg [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="200" height="100">
```
Comment (RazrFalcon) [inline docs/images/after/output-newline.svg]: Does "before" image is missing?

File: src/main.rs (hunk 1 above)
Comment (aaronfranke) [reply src/main.rs]: Yes, that works. I originally tried doing this for `WriteOptions` but was told it's not a member.

File: docs/images/after/output-newline.svg (hunk 3 above)
Comment (aaronfranke) [reply docs/images/after/output-newline.svg]: I'm re-using another example's after image as the before image, but if desired I can copy it.

File: src/main.rs [hunk 4]
```diff
@@ -174,6 +174,13 @@ fn main() {
         }
     }
 
+    // Optionally add a newline to the end of the file.
+    // This is placed after the check for if the file is smaller. It's OK if the
+    // file grows when adding a newline, since the user explicitly wanted that.
+    if cleaning_opt.append_newline {
+        buf.append(&mut vec![10]); // 10 is the decimal value of the newline character.
```
Comment (RazrFalcon) [inline src/main.rs]: This should be replaced with `buf.push(b'\n');`  and we are ready to merge.

**Comment by RazrFalcon** [comment]
Thanks again!

**Pull request merged**